### DISCUSSION TITLE
I added these helper functions to my ~/.visidatarc today

### DISCUSSION THREAD

**Discussion by ghost** [saulpw/visidata#1117] [Show and tell]
I needed to build some new columns today, so I added the following functions to my `~/.visidatarc`:

``` python
def IF(case, a, b):
    """If case evaluates to True, return a, else return b"""
    if case:
        return a
    else:
        return b


def FREQUENCY(items: list) -> dict:
    """Return dict of frequencies based on list of items"""
    from collections import defaultdict
    frequencies = defaultdict(int)
    for item in items:
        frequencies[item] += 1
    return frequencies
```

You can use these functions for example with expression columns using `=`:

|col1  |col2  |col3  |
|------|------|------|
|1     |4     |3     |
|4     |4     |4     |
|7     |8     |9     |

e.g. Check if a row contains the number '4' at least 2 times (Make sure the cols are numeric!):

`=FREQUENCY([col1, col2, col3])[4] >= 2` yields the following:

|col1  |col2  |col3  |FREQUENCY\(\[col1, col2, col3\]\)\[4\] \>= 2|
|-----:|-----:|-----:|---------------------------------------|
|1     |4     |3     |False                                  |
|4     |4     |4     |True                                   |
|7     |8     |9     |False                                  |

OR, Use two different summary values based on a condition:

`=IF(col2 < col3, '😎', '😥')`

|col1  |col2  |col3  |IF\(col2 < col3, '😎', '😥'\)  |
|-----:|-----:|-----:|-----------------------------|
|1     |4     |3     |😥                            |
|4     |4     |4     |😥                            |
|7     |8     |9     |😎                            |

**Comment by saulpw** [comment]
Really cool, @davidwales!  I can see these being useful for people coming from Excel.  For the `FREQUENCY`, what about using `collections.Counter` (might even be a drop-in replacement)?

**Reply by ghost** [reply]
Thanks @saulpw! `collections.Counter` does exactly the same thing... 😮
Here's my new version:

``` python
def FREQUENCY(items):
    """Return collections.Counter of frequencies based on list of items"""
    from collections import Counter
    return Counter(items)
```

I did it as a function so I can defer the import. Probably only saves a millisecond though!
If you don't care about deferring the import:

``` python
from collections import Counter
FREQUENCY = Counter
```

Or if you don't care about the name `FREQUENCY`, just:

``` python
from collections import Counter
```

**Reply by saulpw** [reply]
As long as we're riffing on this, how about `from collections import Counter as FREQUENCY`?

**Reply by ghost** [reply]
That's great. I should know this. I'm always typing `import pandas as pd`...

**Comment by frosencrantz** [comment]
That is cool @davidwales .  I think another function that might be interesting is `VLOOKUP` , where you could map data from another sheet, an alternative to doing a join.

**Comment by geekscrapy** [comment]
Personally, I found the following useful:
```
from collections import Counter
vd.aggregator('counter', Counter, 'count per value')
```

**Reply by ghost** [reply]
Is there a way to use aggregators across columns? Or do they only work on rows?

**Reply by saulpw** [reply]
They only work on rows within a single column.  But you can convert columns into rows; see the Melt command.
